xarray_beam/_src/core_test.py

```diff
@@ -456,15 +456,22 @@ def test_validate(self):
     with self.assertRaisesWithLiteralMatch(
         ValueError,
         (
-            'inconsistent dataset sizes: '
-            "[Frozen({'x': 3, 'y': 6, 'z': 10}),"
-            " Frozen({'x': 3, 'y': 6})]"
+            'inconsistent dataset sizes, only the first can be greater: '
+            "[Frozen({'x': 3, 'y': 6}),"
+            " Frozen({'x': 3, 'y': 6, 'z': 10})]"
         ),
     ):
       test_util.EagerPipeline() | xbeam.DatasetToChunks(
-          [dataset, dataset.drop_dims('z')]
+          [dataset.drop_dims('z'), dataset], chunks={'x': 1}
       )
 
+    try:
+      test_util.EagerPipeline() | xbeam.DatasetToChunks(
+        [dataset, dataset.drop_dims('y')], chunks={'x': 1}
```

Comment:
`drop_dims` drops all variables that include the dimension, so the first dataset has variables ["foo", "bar", "baz"] and the second only has ["bar", "baz"].

The case we care about supporting is when both datasets have the same data variables, but the some variables on later datasets have fewer dimensions, e.g., `[dataset, dataset.isel(y=0, drop=True)]`